slight/src/commands/run.rs

```diff
@@ -33,32 +34,54 @@ pub async fn handle_run(module: impl AsRef<Path>, toml_file_path: impl AsRef<Pat
     let toml_file_contents = std::fs::read_to_string(&toml_file_path)?;
     let toml = toml::from_str::<TomlFile>(&toml_file_contents)?;
 
+    let mut linked_capabilities: HashSet<String> = HashSet::new();
+    let mut capability_store: HashMap<String, BasicState> = HashMap::new();
+
     tracing::info!("Starting slight");
 
     let resource_map = Arc::new(Mutex::new(StateTable::default()));
 
-    let host_builder = build_store_instance(&toml, &toml_file_path, resource_map.clone(), &module)?;
+    let host_builder = build_store_instance(
+        &toml,
+        &toml_file_path,
+        &mut linked_capabilities,
+        &mut capability_store,
+        resource_map.clone(),
+        &module,
+    )?;
     let (mut store, instance) = host_builder.build().await?;
 
     let caps = toml.capability.as_ref().unwrap();
     // looking for events capability.
-    let events_enabled = caps.iter().any(|cap| cap.name == "events");
+    let events_enabled = caps.iter().any(|cap| cap.resource == "events");
 
     // looking for http capability.
-    let http_enabled = caps.iter().any(|cap| cap.name == "http");
+    let http_enabled = caps.iter().any(|cap| cap.resource == "http");
 
     if events_enabled {
         log::debug!("Events capability enabled");
-        let guest_builder =
-            build_store_instance(&toml, &toml_file_path, resource_map.clone(), &module)?;
+        let guest_builder = build_store_instance(
+            &toml,
+            &toml_file_path,
+            &mut linked_capabilities,
```

Comment:
If you pass the mutable reference of `linked_capabilities` and `capability_store` to events and http buidler, that will effectively make it link none of the capabilities, because the hashset is not empty after line 44. How about `build_store_instance` creates `linked_capability` and `capability_store` variables locally?

Reply:
done ✅